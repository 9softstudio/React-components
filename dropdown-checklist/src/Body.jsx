import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Filter from './Filter';
import Scroller from './Scroller';

export default class Body extends Component {
    render() {
        const { options, listVisible, normalizedData, searchValue } = this.props;
        const { showFilter, filterDelay, cssClass} = options;
        const width = options.width ? options.width :  this.props.dropdownElement.offsetWidth

        return (
            <div className={cssClass + " fdcl__dropdown"}>
                {showFilter ? <Filter width={width} onFilterChange={this.props.onFilterChange} filterDelay={filterDelay} searchValue={searchValue} /> : ""}
                <Scroller options={options} 
                          normalizedData={normalizedData} 
                          width={width}
                          onCheckChanged = {this.props.onCheckChanged}
                          onExpandClick = {this.props.onExpandClick}/>
            </div>
        );
    }
};