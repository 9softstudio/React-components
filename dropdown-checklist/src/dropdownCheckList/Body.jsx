import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Filter from './Filter';

export default class Body extends Component {
    getScrollerElement = () => {
        var { height, cssClass, width, mode, dropdownName, filterDelay } = this.props.options;
        var { dropdownElement, normalizedData } = this.props;

        width = width ? width : dropdownElement.offsetWidth;
        var scrollerElement = <div className={height ? "fdcl__scroller" : ""} style={{ height: height, width: width }}>
                                {this.createListElement(normalizedData)}
                            </div>;

        return scrollerElement;
    }

    createListElement = (items) => {
        var elements = [];

        for (var i = 0; i < items.length; i++) {
            elements.push(this.createListItemElement(items[i]));
        }

        return <ul className="ddcl">{elements}</ul>;
    }

    createListItemElement = (item) => {
        var { singleSelect, dataKeyName, listItemClassName, dropdownName } = this.props.options;

        var levelClassName = " level-" + item.level;
        var singleClassName = singleSelect === item.level ? " single " : "";
        var expandedClassName = item.expanded ? " expanded " : "";
        var rootClassName = item[dataKeyName] === 0 ? " root " : "";

        var expandElement;
        var childElements;

        if (item.items && item.items.length > 0) {
            let attributeKey = { 'data-key': item[dataKeyName] }
            expandElement = <div className="ddcl__expand" {...attributeKey} onClick={this.onExpandClick}></div>
            childElements = this.createListElement(item.items);
        }

        return <li key={item[dataKeyName]} className={listItemClassName + expandedClassName + singleClassName + rootClassName + levelClassName}>
            {this.createCheckItemElement(item)}
            {expandElement}
            {childElements}
        </li>;
    }

    createCheckItemElement = (item) => {
        var { singleSelect, selectableLevel, itemCheckClassName, dataKeyName } = this.props.options;

        var inputChild = "";
        if ((!singleSelect || singleSelect <= item.level) &&
            (selectableLevel === -1 || item.level <= selectableLevel)) {

            let inputProps = {
                checked: item.checked,
                onChange: this.onCheckChanged,
                className: itemCheckClassName,
                type: "checkbox",
                value: item[dataKeyName]
            };

            inputChild = <input {...inputProps} />;
        }

        return <label className="ddci">{inputChild}<div className="ddci__text">{item.text}</div></label>;

    }

    //#Region Utilities
    onCheckChanged = (e) => {
       this.props.onCheckChanged(e);
    }

    onExpandClick = (e) => {
        this.props.onExpandClick(e);
    } 

    onFilterChange = (value) => {
        this.props.onFilterChange();
    }
    //#EndRegion Utilities

    render() {
        var { options, listVisible } = this.props;
        var { showFilter, filterDelay, cssClass, width } = options;
        return (
            <div className={cssClass + " fdcl__dropdown"}>
                {showFilter ? <Filter width={width} onFilterChange={this.onFilterChange} filterDelay={filterDelay} /> : ""}
                {this.getScrollerElement()}
            </div>
        );
    }
};