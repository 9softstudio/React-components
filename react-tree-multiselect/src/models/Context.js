import React from 'react';

export default React.createContext({
    rootNode: {},
    nodes: [],
    onToggle: () => { },
    onCheck: () => { }
});