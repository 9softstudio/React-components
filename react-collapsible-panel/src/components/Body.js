import React from 'react';
import { mergeClassName } from '../utils';
import Context from '../panel-context';

export default (props) => {
    return (<Context.Consumer>{({ isExpanded  }) => isExpanded && (
        <div className={mergeClassName(props, 'la-panel__body')}>
            {props.children}
        </div>
    )}</Context.Consumer>);
};