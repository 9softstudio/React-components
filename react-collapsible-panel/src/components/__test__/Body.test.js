import React from 'react';
import Body from '../Body';
import Context from '../../panel-context';
import renderer from 'react-test-renderer';

describe('Panel Body is expanded:', () => {
    test('renders correctly', () => {
        const component = renderer
            .create(
                <Context.Provider value={({ isExpanded: true })}>
                    <Body />
                </Context.Provider>
            );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('custom class name', () => {
        const component = renderer
            .create(
                <Context.Provider value={({ isExpanded: true })}>
                    <Body className="hello" />
                </Context.Provider>
            );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('has children', () => {
        const component = renderer
            .create(
                <Context.Provider value={({ isExpanded: true })}>
                    <Body className="hello">
                        <span>world</span>
                    </Body>
                </Context.Provider>
            );
        expect(component.toJSON()).toMatchSnapshot();
    });
});

test('Panel Body is collapsed: not render', () => {
    const component = renderer
        .create(
            <Context.Provider value={({ isExpanded: false })}>
                <Body className="hello">
                    <span>world</span>
                </Body>
            </Context.Provider>
        );
    expect(component.toJSON()).toMatchSnapshot();
});