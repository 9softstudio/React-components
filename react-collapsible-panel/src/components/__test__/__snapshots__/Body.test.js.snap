// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel Body is collapsed: not render 1`] = `null`;

exports[`Panel Body is expanded: custom class name 1`] = `
<div
  className="hello la-panel__body"
/>
`;

exports[`Panel Body is expanded: has children 1`] = `
<div
  className="hello la-panel__body"
>
  <span>
    world
  </span>
</div>
`;

exports[`Panel Body is expanded: renders correctly 1`] = `
<div
  className="la-panel__body"
/>
`;
